services/user_service.py

```
@@ -0,0 +1,67 @@
+
+            file_content = await profile_image.read()
+            if len(file_content) > 5 * 1024 * 1024: #5MB
+                raise HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST,
+                    detail="파일의 용량은 5MB를 넘을 수 없습니다."
+                )
+
+            file_name = f"{uuid.uuid4()}_{profile_image.filename}"
+            file_path = os.path.join(UPLOAD_DIR, file_name)
+
+            with open(file_path, "wb") as f:
```

Ruff (0.14.11)

Async functions should not open files with blocking methods like `open`

(ASYNC230)